File: QoRManager/Shared/src/Network/Connection/IClient.java
package Network.Connection;

import com.google.protobuf.Message;

import java.io.IOException;

public interface IClient extends ICommunication {
    boolean sendMessage(String message) throws IOException, NullPointerException;

    <T extends Message> boolean sendMessage(T message) throws IOException, NullPointerException;

    String sendMessageWithResponse(String message) throws IOException, NullPointerException;
}

File: Encapsulation/CommonLibrary/Communication/Connection/IClient.cs
using CommonLibrary.Communication.DataModel;
using Google.Protobuf;

namespace CommonLibrary.Communication.Connection
{
    public interface IClient
    {
        Task<bool> SendProtobufToOtherAsync<T>(T message, ConnectionInformation connectionInformation) where T : IMessage;

        Task<bool> SendMessageAsync(byte[] message, ConnectionInformation connectionInformation);
    }
}
